src/nvim/ui_client.c

```diff
@@ -56,33 +56,30 @@ void ui_client_run(bool remote_ui)
 
   loop_poll_events(&main_loop, 1);
 
-  Array args = ARRAY_DICT_INIT;
-  Dictionary opts = ARRAY_DICT_INIT;
-
-  PUT(opts, "rgb", BOOLEAN_OBJ(true));
-  PUT(opts, "ext_linegrid", BOOLEAN_OBJ(true));
-  PUT(opts, "ext_termcolors", BOOLEAN_OBJ(true));
-
+  MAXSIZE_TEMP_ARRAY(args, 3);
+  ADD_C(args, INTEGER_OBJ(Columns));
+  ADD_C(args, INTEGER_OBJ(Rows));
+
+  MAXSIZE_TEMP_DICT(opts, 9);
```

Comment:
off-topic nit, but didn't notice until this PR: since these MAXSIZE_ macros are declaring vars, might be helpful to rename them like:
```suggestion
  INIT_TEMP_DICT(opts, 9);
```

(the "MAXSIZE" prefix doesn't seem worth mentioning in the name)